app/services/bulk_upload/lettings/year2024/csv_parser.rb

```diff
@@ -0,0 +1,113 @@
+require "csv"
+
+class BulkUpload::Lettings::Year2024::CsvParser
+  include CollectionTimeHelper
+
+  FIELDS = 130
+  MAX_COLUMNS = 142
+  FORM_YEAR = 2024
+
+  attr_reader :path
+
+  def initialize(path:)
+    @path = path
+  end
+
+  def row_offset
+    if with_headers?
+      rows.find_index { |row| row[0].match(/field number/i) } + 1
+    else
+      0
+    end
+  end
+
+  def col_offset
+    with_headers? ? 1 : 0
+  end
+
+  def cols
+    @cols ||= ("A".."EL").to_a
```

Comment:
similarly, if there are fewer fields we shouldn't go all the way to EL right?

Reply:
You're right! It looks like it only goes up to EA!